docs/configuration.md

```diff
@@ -449,13 +449,41 @@ Alternatively, pass the notebook file(s) to `ruff` on the command-line directly.
 
 ## Command-line interface
 
-Some configuration options can be provided via the command-line, such as those related to rule
-enablement and disablement, file discovery, logging level, and more:
+All configuration options can be provided or overridden via the command-line.
+Some of the most common have their own dedicated command-line flag.
+This includes the configuration options relating to rule enablement and disablement,
+file discovery, and logging level:
 
 ```shell
 ruff check path/to/code/ --select F401 --select F403 --quiet
 ```
 
+Configuration options that do not have a dedicated flag can be set or
+overridden using the `--config` flag. The `--config` flag is most often
+used to point to the configuration file that you would like ruff to use,
```

Comment:
Nit: I would explain one concept at a time by

1. Explain the configuration options that can be overriden globally (they also have different semantics)
2. Explain the `--config` option

```suggestion
Some configuration options can be provided or overriden via the command-line, such as those related to rule enablement and disablement, file discovery, logging level, and more:

```shell
ruff check path/to/code/ --select F401 --select F403 --quiet
```

Configuration options that do not have a dedicated flag can be set using the `--config` flag. The `--config` flag is most often
used to point to the configuration file that you would like ruff to use,
```

I think it would be worth explaining in more detail how the config options are overriden. Does `--config line-length=100` override the `line-length` settings for all files or do configuration files in sub-directories have priority (because they're closer to the file)?